feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ConversationAnswerMessagesUITest.kt

```diff
@@ -0,0 +1,228 @@
+        }
+
+        composeTestRule.onNodeWithText(answers[2].content!!, useUnmergedTree = true)
+            .performSemanticsAction(SemanticsActions.OnLongClick)
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves)).performClick()
+
+        assert(resolvedPost != null)
+        assert(resolvedPost is AnswerPostPojo)
+        assert((resolvedPost as AnswerPostPojo).content == answers[2].content!!)
+    }
+
+    @Test
+    fun `test GIVEN post is resolved WHEN un-resolving the post THEN the post is un-resolved`() {
+        val resolvingIndex = 0
+
+        val modifiedAnswers = answers.toMutableList()
+        modifiedAnswers[resolvingIndex] = modifiedAnswers[resolvingIndex].copy(resolvesPost = true)
+        val resolvedPost = post.copy(
+            resolved = true,
+            answers = modifiedAnswers
+        )
+
+        var unresolvedPost: IBasePost? = null
+
+        setupUi(resolvedPost) { post ->
+            unresolvedPost = post
+            CompletableDeferred()
+        }
+
+        composeTestRule.onNodeWithText(answers[resolvingIndex].content!!, useUnmergedTree = true)
+            .performSemanticsAction(SemanticsActions.OnLongClick)
+        composeTestRule.onNodeWithText(context.getString(R.string.post_does_not_resolve))
+            .performClick()
+
+        assert(unresolvedPost != null)
+        assert(unresolvedPost is AnswerPostPojo)
+        assert((unresolvedPost as AnswerPostPojo).content == answers[resolvingIndex].content!!)
+    }
+
+    @Test
+    fun `test GIVEN the post is not resolved and no answer post is resolving THEN the post is shown as not resolved and no answer post is shown as resolving`() {
+        setupUi(post) { CompletableDeferred() }
+
+        composeTestRule.onNodeWithText(post.content).assertExists()
+        for (answer in answers) {
+            composeTestRule.onNodeWithText(answer.content!!).assertExists()
+        }
+        composeTestRule.onNodeWithText(context.getString(R.string.post_is_resolved))
+            .assertDoesNotExist()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves))
+            .assertDoesNotExist()
+    }
+
+    @Test
+    fun `test GIVEN the post is resolved and one answer post is marked as resolving THEN the post is shown as resolved and this answer post is shown as resolving`() {
+        val resolvingIndex = 0
+
+        val modifiedAnswers = answers.toMutableList()
+        modifiedAnswers[resolvingIndex] = modifiedAnswers[resolvingIndex].copy(resolvesPost = true)
+        val resolvedPost = post.copy(
+            resolved = true,
+            answers = modifiedAnswers
+        )
+
+        setupUi(resolvedPost) { CompletableDeferred() }
+
+        composeTestRule
+            .onNodeWithTag(testTagForAnswerPost(answers[resolvingIndex].postId), useUnmergedTree = true)
+            .assert(hasAnyChild(hasText(context.getString(R.string.post_resolves))))
+        composeTestRule
```

Comment:
please convert this to a for loop to avoid code duplication

Reply:
The second one is negated with `.not()`. I think we can't put them both into a for loop that way.

Reply:
I did not run this code but something like this should work. 

```kotlin
 val resolvesAssertion = hasAnyChild(hasText(context.getString(R.string.post_resolves)))

for (i in answers.indices) {
    composeTestRule
        .onNodeWithTag(TEST_TAG_THREAD_LIST, useUnmergedTree = true)
        .performScrollToIndex(i)

    composeTestRule
        .onNodeWithTag(testTagForAnswerPost(answers[i].postId), useUnmergedTree = true)
        .assert(if (i == resolvingIndex) resolvesAssertion else resolvesAssertion.not())
}
```

Reply:
Works perfectly fine. Thanks for your help :)